jerry-core/jerry.cpp

```diff
@@ -430,6 +431,104 @@ jerry_api_create_object (void)
 } /* jerry_api_create_object */
 
 /**
+ * Create an array object
+ *
+ * Note:
+ *      caller should release the object with jerry_api_release_object, just when the value becomes unnecessary.
+ *
+ * @return pointer to created array object
+ */
+jerry_api_object_t *
+jerry_api_create_array_object (jerry_api_size_t size) /* size of array */
+{
+  JERRY_ASSERT (size > 0);
+
+  ecma_number_t *length_num_p = ecma_alloc_number ();
+  *length_num_p = ecma_uint32_to_number (size);
+  ecma_value_t array_length = ecma_make_number_value (length_num_p);
+
+  jerry_api_length_t argument_size = 1;
+  ecma_completion_value_t get_completion = ecma_op_create_array_object (&array_length, argument_size, true);
+  ecma_value_t val = ecma_get_completion_value_value (get_completion);
+  jerry_api_object_t *obj_p = ecma_get_object_from_value (val);
+
+  if (!ecma_is_completion_value_normal (get_completion))
+  {
+    JERRY_ASSERT (ecma_is_completion_value_throw (get_completion));
+  }
+
+  ecma_free_value (array_length, true);
+  return obj_p;
+} /* jerry_api_create_array_object */
+
+/**
+ * Set value of field in the specified array object
+ *
+ * @return true, if field value was set successfully
+ *         false - otherwise.
+ */
+bool
+jerry_api_set_array_index_value (jerry_api_object_t *array_obj_p, /* array object */
+                                 jerry_api_length_t index, /* index to be written */
+                                 jerry_api_value_t *value_p) /* value to set*/
+{
+  bool is_successful = true;
+  ecma_string_t *str_idx_p = ecma_new_ecma_string_from_uint32 ((uint32_t) index);
+  ecma_value_t value;
+  jerry_api_convert_api_value_to_ecma_value (&value, value_p);
+  ecma_completion_value_t get_completion = ecma_op_object_put (array_obj_p, str_idx_p, value, false);
+
+  if (!ecma_is_completion_value_normal (get_completion))
+  {
+    JERRY_ASSERT (ecma_is_completion_value_throw (get_completion));
```

Comment:
If `is_throw` argument of `ecma_op_object_put` is `false`, exception can't be thrown by the function.
In the case, the `if` block could also be replaced with assertion that completion value is normal.